Branch-SDK/src/io/branch/referral/indexing/BranchUniversalObject.java

```diff
@@ -0,0 +1,642 @@
+        Branch branchInstance = Branch.getInstance();
+        if (branchInstance != null && branchInstance.getLatestReferringParams() != null) {
+            JSONObject latestParam = branchInstance.getLatestReferringParams();
+            try {
+                if (latestParam.has("+clicked_branch_link") && latestParam.getBoolean("+clicked_branch_link")) {
+                    branchUniversalObject = new BranchUniversalObject();
+
+                    if (latestParam.has(Defines.Jsonkey.ContentTitle.getKey())) {
+                        branchUniversalObject.title_ = latestParam.getString(Defines.Jsonkey.ContentTitle.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentTitle.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.CanonicalIdentifier.getKey())) {
+                        branchUniversalObject.canonicalIdentifier_ = latestParam.getString(Defines.Jsonkey.CanonicalIdentifier.getKey());
+                        latestParam.remove(Defines.Jsonkey.CanonicalIdentifier.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentKeyWords.getKey())) {
+                        JSONArray keywordJsonArray = latestParam.getJSONArray(Defines.Jsonkey.ContentKeyWords.getKey());
+                        for (int i = 0; i < keywordJsonArray.length(); i++) {
+                            branchUniversalObject.keywords_.add((String) keywordJsonArray.get(i));
+                        }
+                        latestParam.remove(Defines.Jsonkey.ContentKeyWords.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentDesc.getKey())) {
+                        branchUniversalObject.description_ = latestParam.getString(Defines.Jsonkey.ContentDesc.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentDesc.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentImgUrl.getKey())) {
+                        branchUniversalObject.imageUrl_ = latestParam.getString(Defines.Jsonkey.ContentImgUrl.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentImgUrl.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentType.getKey())) {
+                        branchUniversalObject.type_ = latestParam.getString(Defines.Jsonkey.ContentType.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentType.getKey());
+                    }
+                    if (latestParam.has(Defines.Jsonkey.ContentExpiryTime.getKey())) {
+                        branchUniversalObject.expirationInMilliSec_ = latestParam.getLong(Defines.Jsonkey.ContentExpiryTime.getKey());
+                        latestParam.remove(Defines.Jsonkey.ContentExpiryTime.getKey());
+                    }
+
+                    ///-----------Link Properties----------------///
```

Comment:
how about the other parameters added by back end , such as params starting with  `~`, `$` and `+` . Currently we are removing them after setting their respective value to BOU. Do we need keep them also in BOU#ContentMetadata_